lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -2538,27 +2563,73 @@ def wallet(self):
     def now(self):
         return self.daemon._get_long_count_timestamp()
 
-    def _add_txid(self, txid):
-        self.name_data['txid'] = txid
-        return defer.succeed(None)
+    def _add_txid(self):
+        def _save(txid):
+            assert txid is not None
+            self.name_data['txid'] = txid
+            return txid
+
+        d = self.wallet.get_txid_for_name(self.name)
+        d.addCallback(_save)
+        return d
+
+    def _confirm_should_remove(self):
+        def _remove_if_should(lbry_file):
+            if lbry_file:
+                d = _get_and_check_tx(lbry_file.txid)
+                d.addCallback(lambda r: None if not r else self._remove_abandoned(lbry_file))
+                return d
+            return defer.succeed(None)
+
+        def _get_and_check_tx(txid):
+            d = self.wallet.get_transaction(txid)
+            d.addCallbacks(_check_tx, lambda _: False)
+            return d
+
+        def _check_tx(tx):
+            confirms = tx['confirmations']
+            if confirms > 6:
+                return True
+            return False
+        lbry_file = next(lf for lf in self.daemon.lbry_file_manager.lbry_files if lf.uri == self.name)
+        d = _remove_if_should(lbry_file)
```

Comment:
I think we need to revert ResolveNameHelper class back to what it was before, after reviewing I don't think the changes are relevant to this patch.